lib/widgets/inbox.dart

```diff
@@ -284,16 +286,32 @@ abstract class _HeaderItem extends StatelessWidget {
           const SizedBox(width: 5),
           Expanded(child: Padding(
             padding: const EdgeInsets.symmetric(vertical: 4),
-            child: Text(
-              style: TextStyle(
-                fontSize: 17,
-                height: (20 / 17),
-                // TODO(design) check if this is the right variable
-                color: designVariables.labelMenuButton,
-              ).merge(weightVariableTextStyle(context, wght: 600)),
+            child: RichText(
               maxLines: 1,
               overflow: TextOverflow.ellipsis,
-              title(zulipLocalizations)))),
+              text: TextSpan(
+                children: [
+                  TextSpan(
+                    text: title(zulipLocalizations),
+                    style: TextStyle(
+                      fontSize: 17,
+                      height: 20 / 17,
+                      // TODO(design) check if this is the right variable
+                      color: designVariables.labelMenuButton,
+                    ).merge(weightVariableTextStyle(context, wght: 600))),
+                  if (isArchived)
+                    // TODO(#1285): Avoid concatenating translated strings
+                    WidgetSpan(
+                      child: Padding(
+                        padding: EdgeInsetsDirectional.only(start: 4),
+                        child: Text(
+                          zulipLocalizations.channelArchivedLabel,
+                          style: TextStyle(
+                            fontSize: 17,
+                            height: 20 / 17,
+                            color: MessageListTheme.of(context).streamRecipientHeaderChevronRight,
```

Comment:
This line is too long. Let's just extract the `MessageListTheme` as a variable at the start.

Reply:
It also doesn't seem right that we are using `streamRecipientHeaderChevronRight` as the color here. Web uses `--color-text-message-header-archived`; how about adding that to `DesignVariables`'s "Not named variables in Figma; taken from older Figma drafts, or elsewhere." section

Let's also have a TODO(design) comment to check if this is the right variable.